shim.h

```diff
@@ -179,6 +179,7 @@
 #include "include/simple_file.h"
 #include "include/str.h"
 #include "include/tpm.h"
+#include "include/tdx.h"
```

Comment:
I think vathpela's comment about this still stands.

Reply:
@frozencemetery I hope we can separate the TDX and TPM's definition (header file) and implementation (C file), because:

- TDX([Trusted Domain Extension](https://software.intel.com/content/www/cn/zh/develop/articles/intel-trust-domain-extensions.html)) is designed for confidential VM. TPM is major designed for TPM device on the baremetal host.
- TDX use similiar structure on measurement like TPM, but not exact same according to recent update https://www.mail-archive.com/devel@edk2.groups.io/msg36488.html. So even put the TDX definition in TPM header file, the actual source code could not be reused much.
- Grub2 provide grub_file_verifier framework https://github.com/rhboot/grub2/blob/master/grub-core/commands/tpm.c#L80, you can register TPM, TDX, or other approach to achieve file verifier functionality. In today's shim, there is no such framework, so I have to call tdx_log_measurement() within tpm.c, which looks not good, but sound like there is no other good way.